src/GT.Core/Services/Impl/GTListingService.cs

```diff
@@ -1,9 +1,277 @@
-﻿using GT.Core.Services.Interfaces;
+﻿using GT.Core.DTO.Impl;
+using GT.Core.FilterModels.Impl;
+using GT.Core.FilterModels.Interfaces;
+using GT.Core.Services.Interfaces;
+using GT.Data.Data.GTAppDb.Entities;
+using GT.Data.Repositories.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
 
 namespace GT.Core.Services.Impl
 {
+	/// <summary>
+	/// Contains business logic for listing objects. Used to map and convert data transfer objects and entities.
+	/// </summary>
 	public class GTListingService : IGTListingService
 	{
+		private readonly ILogger<GTListingService> _logger;
+		private readonly IGTCompanyService _companyService;
+		private readonly IGTExperienceLevelService _experienceLevelService;
+		private readonly IGTLocationService _locationService;
+		private readonly IGTGenericRepository<Listing> _listingRepository;
+		private readonly IGTGenericRepository<Company> _companyRepository;
+		private readonly IGTGenericRepository<Location> _locationRepository;
+		private readonly IGTGenericRepository<ExperienceLevel> _experienceLevelRepository;
 
+		public GTListingService(
+			ILogger<GTListingService> logger,
+			IGTCompanyService companyService,
+			IGTExperienceLevelService experienceLevelService,
+			IGTLocationService locationService,
+			IGTGenericRepository<Listing> listingRepository,
+			IGTGenericRepository<Company> companyRepository,
+			IGTGenericRepository<Location> locationRepository,
+			IGTGenericRepository<ExperienceLevel> experienceLevelRepository)
+		{
+			_logger = logger;
+			_companyService = companyService;
+			_experienceLevelService = experienceLevelService;
+			_locationService = locationService;
+			_listingRepository = listingRepository;
+			_companyRepository = companyRepository;
+			_locationRepository = locationRepository;
+			_experienceLevelRepository = experienceLevelRepository;
+		}
+
+		/// <summary>
+		/// Converts a DTO to entities and updates the database. 
+		/// Requires the signed in users Id for assignment of CreatedBy property.
+		/// </summary>
+		/// <param name="listingDTO"></param>
+		/// <returns>The input DTO with an updated Id.</returns>
+		public async Task<ListingDTO> AddAsync(ListingDTO listingDTO, string signedInUserId)
+		{
+			if (listingDTO is null)
+			{
+				_logger.LogError($"Attempted to add a null reference to the database.");
+				return null;
+			}
+
+			// Map DTO to entity
+
+			var newListing = await CreateListingEntity(listingDTO, signedInUserId);
+
+			// Update database
+			await _listingRepository.AddAsync(newListing);
+
+
+			// Map updates to DTO
+			listingDTO.Id = newListing.Id;
+
+			// return the updated ListingDTO
+			return listingDTO;
+		}
+
+		public async Task<Listing> CreateListingEntity(ListingDTO listingDTO, string signedInUserId)
+		{
+			var newListing = new Listing()
```

Comment:
Properties would look better declared within "{ };"

Reply:
This will be managed in issue #64.